legacy/firmware/ethereum.c

```diff
@@ -37,14 +37,16 @@
 #include "transaction.h"
 #include "util.h"
 
-/* maximum supported chain id.  v must fit in an uint32_t. */
-#define MAX_CHAIN_ID 2147483629
+/* maximum supported chain id.
+  signature_v must also fit in an uint64_t.
+  */
+#define MAX_CHAIN_ID ((ULLONG_MAX - 35) >> 1)
```

Comment:
Let's not use the macro which value depends on used architecture:

```suggestion
#define MAX_CHAIN_ID ((0xFFFFFFFFFFFFFFFF - 35) >> 1)
```

Reply:
:+1: